api/operator/v1alpha1/istiocsr_types.go

```diff
@@ -194,7 +194,7 @@ type IstiodTLSConfig struct {
 	// This field can have a maximum of 63 characters.
 	// +kubebuilder:validation:MinLength:=1
 	// +kubebuilder:validation:MaxLength:=63
-	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="trustDomain must consist of lowercase alphanumeric characters, hyphens ('-'), and periods ('.'). Each block, separated by periods, must start and end with an alphanumeric character. Hyphens are not allowed at the start or end of a block, and consecutive periods are not permitted."
+	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$`
```

Comment:
I think it's better to remove regex. We don't have efforts at this stage to test this elaborately. We can enhance the API in future releases.